UnityProject/Assets/Scripts/Systems/Chemistry/Effects/ReleaseGas.cs

```diff
@@ -0,0 +1,33 @@
+
+using System;
+using ScriptableObjects.Atmospherics;
+using Systems.Atmospherics;
+using UnityEngine;
+
+namespace Chemistry.Effects
+{
+
+	[Serializable]
+	[CreateAssetMenu(fileName = "ReleaseGas", menuName = "ScriptableObjects/Chemistry/Effect/ReleaseGas")]
+	public class ReleaseGas : Chemistry.Effect
+	{
+		public GasSO ToRelease;
+		public float AmountToRelease = 10;
+		public float TemperatureK = 293.15f;
+
+		public override void Apply(MonoBehaviour onObject, float amount)
+		{
+			var Matrix =  onObject.gameObject.OnMatrixRoot();
+
+
+			var	metaNode = Matrix.MetaDataLayer.Get(onObject.transform.localPosition.RoundToInt());
+
+			lock (metaNode.GasMix.GasesArray) //no Double lock
+			{
+				var mix = new GasMix(2.5f, TemperatureK);
```

Comment:
why 2.5f?

Reply:
yeah, make it a field and serialize it.

Reply:
volume doesn't change the effect on the  gas at all

Reply:
it is the amount of gas this effect will release, isn't it? Sounds reasonable to make it a serialized field so you can have different amounts.

Reply:
Well you got the temp and moles you don't need to add any more

Reply:
information since, you've got the number of atoms of gas + the energy that the gas has, 
and it's getting added to the local environment so it's fine